tests/cases/fourslash/codeFixInferFromFunctionThisUsageObjectPropertyShorthand.ts

```diff
@@ -0,0 +1,20 @@
+/// <reference path='fourslash.ts' />
+
+// @noImplicitThis: true
+////function returnThisMember([| |]) {
+////     return this.member;
+//// }
+////
+//// interface Container {
+////     member: string;
+////     returnThisMember(): string;
+//// }
+////
+//// const container: Container = {
```

Comment:
what happens where there is no type annotation here?

Reply:
We get this bundle of joy:

```ts
this: { member: string; returnThisMember: () => any; }
```

I'll add a test case.

Reply:
Yeah, I think that's because we have to resolve the function type before we are done getting the type of `this`. It's a pretty common problem with `this` in object literals.